xrpl/models/transactions/transaction.py

```diff
@@ -423,7 +421,7 @@ def get_hash(self: Transaction) -> str:
 
     @classmethod
     def get_transaction_type(
-        cls: Type[Transaction], transaction_type: str
+        cls: Type[Self], transaction_type: str
     ) -> Type[Transaction]:
```

Comment:
```suggestion
    ) -> Type[Self]:
```
?

Reply:
That is actually incorrect. If I call `Payment.get_transaction_type`, it's going to return type `Transaction`, not type `Payment`.

Reply:
```
correct_type = cls.get_transaction_type(value["transaction_type"])
```
`cls.get_transaction_type` takes the transaction type passed into the dictionary and uses that model instead. So if you do `Payment.get_transaction_type(accountSetTx)`, that should redirect you to the `AccountSet` model. That's the check that's done here:
https://github.com/XRPLF/xrpl-py/blob/4779d796e6f808c952214e794be5d9d342412e14/xrpl/models/transactions/transaction.py#L355-L361

Reply:
I agree with your explanation.

I wrote some unit tests to observe what happens if `Type[Self]` is returned - https://github.com/ckeshava/xrpl-py/tree/type_self

I suspect `Type[Transaction]` and `Type[Self]` are identical because this method (`get_trasaction_type`) is not overridden by any of the child classes

Reply:
Exactly - which means that `Payment.get_transaction_type` still returns type `Transaction`, which is no longer `Self` (`Self` is now `Payment`).

Reply:
hmm, it makes sense now. thanks